fem/tmop.cpp

```diff
@@ -1336,6 +1336,7 @@ void DiscreteAdaptTC::FinalizeParDiscreteTargetSpec(const ParGridFunction &t)
                                            ptspec_fes->FEColl(), ncomp);
 
    delete tspec_pgf;
+   tspec.HostReadWrite();
```

Comment:
Would it be better to address the issue inside `ParGridFunction(ParFiniteElementSpace *pf, GridFunction *gf);`,
where `gf->GetData()` might be the cause?

Reply:
The error can also be caused by the `double*` cast to use `GridFunction(FiniteElementSpace *f, double *data)`.

Reply:
> The error can also be caused by the `double*` cast to use `GridFunction(FiniteElementSpace *f, double *data)`.

I think this is the issue (the second argument is a `Vector` implicitly cast to `double *`), so it cannot be solved inside the `(Par)GridFunction` constructor.